lm_eval/tasks/score/agi_eval/utils_agieval.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+import os
+from datasets import Dataset
+from typing import Any, Dict, List
+import re
+import numpy as np
+
+from lm_eval.tasks.score import utils
+from lm_eval.tasks.score.utils import prompt_consistency_rate
+from lm_eval.tasks.score.utils import robustness_doc_to_text
+from lm_eval.utils import eval_logger
+
+
+TEMPLATE_FILE_PATH = os.path.join(os.path.dirname(__file__), "prompt_templates.json")
+
+PROMPT_ROBUSTNESS_TEMPLATE_KEY = "prompt_robustness"
+OPTION_ORDER_ROBUSTNESS_TEMPLATE_KEY = "option_order_robustness"
+
+QUESTION_KEY = "query"
+ANSWER_INDEX_KEY = "gold"
+OPTIONS_KEY = "choices"
+
+LABELS = ['A', 'B', 'C', 'D', 'E']
+
+
+def initial_process_docs(doc: Dataset) -> Dataset:
+    """
+    add question_id to the documents
+    """
+
+    bracket_pattern = r'^\([A-E]\)'
+    letter_space = r'^[A-E] '
+    letter_question_space = r'^[A-E]\? '
+    
+    def __process(_doc, idx):
+        if "question" not in _doc:
+            question = _doc[QUESTION_KEY].split(" Answer Choices:")[0]
+            if question.startswith('Q: '):
+                question = question[3:]
+            _doc["question"] = question
+        if "question_id" not in _doc:
+            _doc["question_id"] = idx
+        if "answer_index" not in _doc:
+            _doc["answer_index"] = _doc[ANSWER_INDEX_KEY][0]
+        if "answer" not in _doc:
+            _doc["answer"] = LABELS[_doc["answer_index"]]
+        if "options" not in _doc:
+            prepared_options = []
+            for option in _doc[OPTIONS_KEY]:
+                if re.match(bracket_pattern, option):
+                    prepared_options.append(option[3:])
+                elif re.match(letter_space, option):
+                    prepared_options.append(option[2:])
+                elif re.match(letter_question_space, option):
+                    prepared_options.append(option[3:])
+                else:
+                    prepared_options.append(option)
+            _doc["options"] = prepared_options
+        return _doc
+    return doc.map(__process, with_indices=True)
+
+def prompt_robustness_process_docs(doc: Dataset) -> Dataset:
```

Comment:
please check that each function has docstring + how about adding high-level of what this file is doing right after the license